scripts/install_dependencies.sh

```diff
@@ -42,8 +43,10 @@ make install
 # Step 2: Install OpenMPI with UCX support
 cd $_DEPS_SRC_DIR
 rm -rf ompi
-git clone --recursive $_OMPI_REPO -b $_OMPI_BRANCH
+git clone --recursive $_OMPI_REPO
 cd ompi
+git checkout $_OMPI_COMMIT_HASH
+git submodule update --init --recursive
```

Comment:
Well, let me correct myself, since we do the ```submodule update``` here, @Yiltan is correct that the ```--recursive``` might not be required. But it also shouldn't cause any issues.